### ISSUE TITLE
Problem with Scrubber_editor_demo.py

### ISSUE THREAD

**Issue by aaronayres35** [enthought/traitsui#1066]
Running on wx gives 

<details>

```
$ python Scrubber_editor_demo.py 
01:03:59 PM: Debug: Adding duplicate image handler for 'Windows bitmap file'
01:03:59 PM: Debug: Adding duplicate animation handler for '1' type
01:03:59 PM: Debug: Adding duplicate animation handler for '2' type
01:03:59 PM: Debug: Adding duplicate image handler for 'Windows bitmap file'
01:03:59 PM: Debug: Adding duplicate animation handler for '1' type
01:03:59 PM: Debug: Adding duplicate animation handler for '2' type
01:03:59 PM: Debug: Adding duplicate image handler for 'Windows bitmap file'
01:03:59 PM: Debug: Adding duplicate animation handler for '1' type
01:03:59 PM: Debug: Adding duplicate animation handler for '2' type
Traceback (most recent call last):
  File "/Users/aayres/.edm/envs/wx-env/lib/python3.6/site-packages/traitsui/wx/view_application.py", line 152, in OnInit
    args=self.args,
  File "/Users/aayres/.edm/envs/wx-env/lib/python3.6/site-packages/traitsui/view.py", line 462, in ui
    ui.ui(parent, kind)
  File "/Users/aayres/.edm/envs/wx-env/lib/python3.6/site-packages/traitsui/ui.py", line 246, in ui
    self.rebuild(self, parent)
  File "/Users/aayres/.edm/envs/wx-env/lib/python3.6/site-packages/traitsui/wx/toolkit.py", line 125, in ui_live
    ui_live.ui_live(ui, parent)
  File "/Users/aayres/.edm/envs/wx-env/lib/python3.6/site-packages/traitsui/wx/ui_live.py", line 52, in ui_live
    _ui_dialog(ui, parent, BaseDialog.NONMODAL)
  File "/Users/aayres/.edm/envs/wx-env/lib/python3.6/site-packages/traitsui/wx/ui_live.py", line 88, in _ui_dialog
    BaseDialog.display_ui(ui, parent, style)
  File "/Users/aayres/.edm/envs/wx-env/lib/python3.6/site-packages/traitsui/wx/ui_base.py", line 71, in display_ui
    ui.owner.init(ui, parent, style)
  File "/Users/aayres/.edm/envs/wx-env/lib/python3.6/site-packages/traitsui/wx/ui_live.py", line 203, in init
    sw = panel(ui, window)
  File "/Users/aayres/.edm/envs/wx-env/lib/python3.6/site-packages/traitsui/wx/ui_panel.py", line 272, in panel
    panel, content[0], ui
  File "/Users/aayres/.edm/envs/wx-env/lib/python3.6/site-packages/traitsui/wx/ui_panel.py", line 431, in fill_panel_for_group
    panel, group, ui, suppress_label, is_dock_window, create_panel
  File "/Users/aayres/.edm/envs/wx-env/lib/python3.6/site-packages/traitsui/wx/ui_panel.py", line 585, in __init__
    self.add_groups(content, panel)
  File "/Users/aayres/.edm/envs/wx-env/lib/python3.6/site-packages/traitsui/wx/ui_panel.py", line 729, in add_groups
    sizer.Add(sg_sizer, growable, style, 2)
wx._core.wxAssertionError: C++ assertion "!(flags & (wxALIGN_BOTTOM | wxALIGN_CENTRE_VERTICAL))" failed at /Users/robind/projects/bb2/dist-osx-py36/build/ext/wxWidgets/src/common/sizer.cpp(2114) in DoInsert(): Vertical alignment flags are ignored with wxEXPAND
OnInit returned false, exiting...
```

</details>

and on qt:
<details>

```
$ python Scrubber_editor_demo.py 
Traceback (most recent call last):
  File "Scrubber_editor_demo.py", line 165, in <module>
    demo.configure_traits()
  File "/Users/aayres/.edm/envs/qt-env/lib/python3.6/site-packages/traits/has_traits.py", line 2015, in configure_traits
    args,
  File "/Users/aayres/.edm/envs/qt-env/lib/python3.6/site-packages/traitsui/qt4/toolkit.py", line 232, in view_application
    context, view, kind, handler, id, scrollable, args
  File "/Users/aayres/.edm/envs/qt-env/lib/python3.6/site-packages/traitsui/qt4/view_application.py", line 84, in view_application
    context, view, kind, handler, id, scrollable, args
  File "/Users/aayres/.edm/envs/qt-env/lib/python3.6/site-packages/traitsui/qt4/view_application.py", line 126, in __init__
    args=self.args,
  File "/Users/aayres/.edm/envs/qt-env/lib/python3.6/site-packages/traitsui/view.py", line 462, in ui
    ui.ui(parent, kind)
  File "/Users/aayres/.edm/envs/qt-env/lib/python3.6/site-packages/traitsui/ui.py", line 246, in ui
    self.rebuild(self, parent)
  File "/Users/aayres/.edm/envs/qt-env/lib/python3.6/site-packages/traitsui/qt4/toolkit.py", line 162, in ui_live
    ui_live.ui_live(ui, parent)
  File "/Users/aayres/.edm/envs/qt-env/lib/python3.6/site-packages/traitsui/qt4/ui_live.py", line 43, in ui_live
    _ui_dialog(ui, parent, BaseDialog.NONMODAL)
  File "/Users/aayres/.edm/envs/qt-env/lib/python3.6/site-packages/traitsui/qt4/ui_live.py", line 65, in _ui_dialog
    BaseDialog.display_ui(ui, parent, style)
  File "/Users/aayres/.edm/envs/qt-env/lib/python3.6/site-packages/traitsui/qt4/ui_base.py", line 284, in display_ui
    ui.owner.init(ui, parent, style)
  File "/Users/aayres/.edm/envs/qt-env/lib/python3.6/site-packages/traitsui/qt4/ui_live.py", line 225, in init
    self.add_contents(panel(ui), bbox)
  File "/Users/aayres/.edm/envs/qt-env/lib/python3.6/site-packages/traitsui/qt4/ui_panel.py", line 259, in panel
    panel = _GroupPanel(content[0], ui).control
  File "/Users/aayres/.edm/envs/qt-env/lib/python3.6/site-packages/traitsui/qt4/ui_panel.py", line 599, in __init__
    layout = self._add_groups(content, inner)
  File "/Users/aayres/.edm/envs/qt-env/lib/python3.6/site-packages/traitsui/qt4/ui_panel.py", line 680, in _add_groups
    panel = _GroupPanel(subgroup, self.ui).control
  File "/Users/aayres/.edm/envs/qt-env/lib/python3.6/site-packages/traitsui/qt4/ui_panel.py", line 601, in __init__
    layout = self._add_items(content, inner)
  File "/Users/aayres/.edm/envs/qt-env/lib/python3.6/site-packages/traitsui/qt4/ui_panel.py", line 867, in _add_items
    ui, object, name, item.tooltip, None
  File "/Users/aayres/.edm/envs/qt-env/lib/python3.6/site-packages/traitsui/editor_factory.py", line 137, in simple_editor
    description=description,
  File "/Users/aayres/.edm/envs/qt-env/lib/python3.6/site-packages/pyface/base_toolkit.py", line 184, in __init__
    raise NotImplementedError(msg % (toolkit, package, name))
NotImplementedError: the qt4 traitsui.qt4 backend doesn't implement scrubber_editor:_ScrubberEditor
```

</details>

**Comment by aaronayres35** [comment]
The error in the traceback above for wx has been fixed on PR #1067 .  However a segfault still occurs on the mouseover of the simple integer scurried.  The other scrubbers seem to work fine.

**Comment by aaronayres35** [comment]
When I enable fault handler I get the following traceback:
<details>

```
(wx-env)~/Desktop/traitsui/traitsui/examples/demo/Advanced aayres  (fix-scrubber-editor-demo) $ python -Xfaulthandler Scrubber_editor_demo.py 
04:29:52 PM: Debug: Adding duplicate image handler for 'Windows bitmap file'
04:29:52 PM: Debug: Adding duplicate animation handler for '1' type
04:29:52 PM: Debug: Adding duplicate animation handler for '2' type
04:29:52 PM: Debug: Adding duplicate image handler for 'Windows bitmap file'
04:29:52 PM: Debug: Adding duplicate animation handler for '1' type
04:29:52 PM: Debug: Adding duplicate animation handler for '2' type
04:29:52 PM: Debug: Adding duplicate image handler for 'Windows bitmap file'
04:29:52 PM: Debug: Adding duplicate animation handler for '1' type
04:29:52 PM: Debug: Adding duplicate animation handler for '2' type
Fatal Python error: Segmentation fault

Current thread 0x000000010dc17dc0 (most recent call first):
  File "/Users/aayres/Desktop/traitsui/traitsui/wx/scrubber_editor.py", line 467 in _enter_window
  File "/Users/aayres/.edm/envs/wx-env/lib/python3.6/site-packages/wx/core.py", line 2224 in MainLoop
  File "/Users/aayres/Desktop/pyface/pyface/util/guisupport.py", line 115 in start_event_loop_wx
  File "/Users/aayres/Desktop/traitsui/traitsui/wx/view_application.py", line 139 in __init__
  File "/Users/aayres/Desktop/traitsui/traitsui/wx/view_application.py", line 92 in view_application
  File "/Users/aayres/Desktop/traitsui/traitsui/wx/toolkit.py", line 219 in view_application
  File "/Users/aayres/Desktop/traits/traits/has_traits.py", line 2049 in configure_traits
  File "Scrubber_editor_demo.py", line 165 in <module>
Segmentation fault: 11
```

</details>

Also, to be more specific it seems that the segfault specifically occurs if I click into a scrubbers text field, move my mouse out of the field, and then move it back into any other textfield
